### PR TITLE
Vyu/tdi 43232 improve where clause for eltoutput m3

### PR DESCRIPTION
**Pull request by vyu-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-43232

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tELTGreenplumOutput/tELTGreenplumOutput_java.xml [hunk 1]
```diff
@@ -55,10 +55,36 @@
     >
     </PARAMETER>
 
-    <PARAMETER NAME="WHERE_CLAUSE" FIELD="MEMO" NUM_ROW="8" SHOW_IF="(DATA_ACTION == 'UPDATE') OR (DATA_ACTION == 'DELETE')"
-	>
+    <PARAMETER NAME="USE_WHERE_CONDITIONS_TABLE" FIELD="CHECK" NUM_ROW="8"
+		SHOW_IF="(DATA_ACTION == 'UPDATE') OR (DATA_ACTION == 'DELETE')">
+		<DEFAULT>false</DEFAULT>
```
Comment (pyzhou-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTGreenplumOutput/tELTGreenplumOutput_java.xml:60]: Is Pierre's decision to set this option unchecking by default?

Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tELTGreenplumOutput/tELTGreenplumOutput_java.xml:60]: > Is it Pierre's decision to set this option unchecking by default?

Oh, yes. Already confirmed with him.

**Review by pyzhou-talend — APPROVED** [review]
LGTM

File: main/plugins/org.talend.designer.components.localprovider/components/tELTMysqlOutput/tELTMysqlOutput_main.javajet [hunk 2]
```diff
@@ -265,32 +267,55 @@ imports="
...
-		updateQuery_<%=cid %> +=" SET <%=updateSetStmt.toString()%> "
+			StringBuffer updateQuery_<%=cid %> = new StringBuffer("UPDATE "+<%=tableName%>+" SET <%=updateSetStmt.toString()%> ");
 <%
-			if(CodeGenerationUtils.hasAlphaNumericCharacter(whereClause)) {
+			if((CodeGenerationUtils.hasAlphaNumericCharacter(whereClause) && (!useWhereTable)) || (useWhereTable && whereConditions.size() > 0)) {
 %>
-		+"  WHERE " + <%=CodeGenerationUtils.replaceAllCrBySpace(whereClause)%>
+				updateQuery_<%=cid %>.append(" WHERE ");
 <%
+				if(!useWhereTable) {
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTMysqlOutput/tELTMysqlOutput_main.javajet:276]: the  if condition should be imporve:
```
if(!useWhereTable) {
    if(CodeGenerationUtils.hasAlphaNumericCharacter(whereClause)){
         // not where table code
    }
}else{
   if(whereConditions.size() > 0){
         // where table code
    }
}
```

Comment (jzhao-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tELTMysqlOutput/tELTMysqlOutput_main.javajet:276]: should check all below code.

Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tELTMysqlOutput/tELTMysqlOutput_main.javajet:276]: > should check all below code.

Thank you very much. I will improve it.

**Review by jzhao-talend — APPROVED** [review]
LGTM

File: main/plugins/org.talend.designer.components.localprovider/components/tELTTeradataOutput/tELTTeradataOutput_main.javajet [hunk 3]
```diff
@@ -290,38 +291,72 @@ try{
...
+			if(whereConditions.size() > 0) {
+%>
+				updateQuery_<%=cid %>.append(" WHERE ");
+<%
+				String operator = "\"\"";
+				for(Map<String, String> whereCondition : whereConditions) {
+					String column_condition = whereCondition.get("COLUMN");
+					String function_condition = whereCondition.get("FUNCTION");
+					String value_condition = whereCondition.get("VALUE_SQL");
+<%
```
Comment (zshi-talendbj) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTTeradataOutput/tELTTeradataOutput_main.javajet:315]: "<%" should use close character

File: main/plugins/org.talend.designer.components.localprovider/components/tELTSybaseOutput/tELTSybaseOutput_main.javajet [hunk 4]
```diff
@@ -141,249 +143,279 @@ imports="
...
+						String function_condition = whereCondition.get("FUNCTION");
+						String value_condition = whereCondition.get("VALUE_SQL");
+%>
+						updateQuery_<%=cid %>.append(<%=operator %> + <%=column_condition %> + " " + <%=function_condition %> + " " + <%=value_condition %>);
+<%
+						operator = "\" AND \"";
+					}
+				}
+			}
+		}
```
Comment (zshi-talendbj) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTSybaseOutput/tELTSybaseOutput_main.javajet:322]: should remove one "}" or will lead compile error

File: main/plugins/org.talend.designer.components.localprovider/components/tELTMysqlOutput/tELTMysqlOutput_main.javajet [hunk 5]
```diff
@@ -265,32 +267,60 @@ imports="
...
+						String function_condition = whereCondition.get("FUNCTION");
+						String value_condition = whereCondition.get("VALUE_SQL");
+%>
+						updateQuery_<%=cid %>.append(<%=operator %> + <%=column_condition %> + " " + <%=function_condition %> + " " + <%=value_condition %>);
+<%
+						operator = "\" AND \"";
+					}
+				}
 			}
+		}
```
Comment (zshi-talendbj) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTMysqlOutput/tELTMysqlOutput_main.javajet:295]: should remove one "}" or will lead compile error

**Pull request merged**